dbt/adapters/snowflake/relation_configs/target_lag.py

```diff
@@ -26,3 +39,101 @@ class SnowflakeDynamicTableTargetLagConfig(RelationConfigBase):
 
     duration: int
     period: SnowflakeDynamicTableTargetLagPeriod
+
+    @property
+    def validation_rules(self) -> Set[RelationConfigValidationRule]:
+        return {
+            RelationConfigValidationRule(
+                validation_check=self.duration is not None and self.period is not None,
+                validation_error=DbtRuntimeError(
+                    "Target lag for a materialized view requires both a duration and a period."
+                ),
+            ),
+            RelationConfigValidationRule(
+                validation_check=(
+                    (
+                        self.duration >= 1
+                        and self.period != SnowflakeDynamicTableTargetLagPeriod.seconds
```

Comment:
`seconds` -> `minute`?

Reply:
And could someone say 0 hours by chance? 🙊

Reply:
I think this is already guarding against the `0 hours` scenario because it's requiring a positive number. But it's a bit hidden since I do a check against the type and then combine it with another check. I'll break this into two checks: positive integer, and >= 60 if your unit is seconds.

Reply:
Oh excellent point about the boolean logic though!